standard/wasi-filesystem/witx/typenames.witx

```diff
@@ -0,0 +1,708 @@
+;; Type names used by low-level WASI interfaces.
+;;
+;; Some content here is derived from [CloudABI](https://github.com/NuxiNL/cloudabi).
+;;
+;; This is a `witx` file. See [here](https://github.com/WebAssembly/WASI/tree/master/docs/witx.md)
+;; for an explanation of what that means.
+
+;;; An array size.
+;;;
+;;; Note: This is similar to `size_t` in POSIX.
+(typename $size (@witx usize))
+
+;;; Non-negative file size or length of a region within a file.
+(typename $filesize u64)
+
+;;; Timestamp in nanoseconds.
+(typename $timestamp u64)
+
+;;; Identifiers for clocks.
+(typename $clockid
+  (enum (@witx tag u32)
+    ;;; The clock measuring real time. Time value zero corresponds with
+    ;;; 1970-01-01T00:00:00Z.
+    $realtime
+    ;;; The store-wide monotonic clock, which is defined as a clock measuring
+    ;;; real time, whose value cannot be adjusted and which cannot have negative
+    ;;; clock jumps. The epoch of this clock is undefined. The absolute time
+    ;;; value of this clock therefore has no meaning.
+    $monotonic
+  )
+)
+
+;;; Error codes returned by functions.
+;;; Not all of these error codes are returned by the functions provided by this
+;;; API; some are used in higher-level library layers, and others are provided
+;;; merely for alignment with POSIX.
+(typename $errno
+  (enum (@witx tag u16)
+    ;;; No error occurred. System call completed successfully.
+    $success
+    ;;; Argument list too long.
+    $2big
```

Comment:
2big?

Reply:
It's `E2BIG` from POSIX, with the `E` removed because that's effectively a namespace prefix in POSIX which is redundant within the enum namespace here.

Reply:
do we have to be the same with posix? starts with digit are werid, prefer toobig or toolong

Reply:
as for some language, enum can not starts  with digit, that's would be a issue for code generation.

Reply:
we already have to special-case this in some code generation tools for that exact reason. I would support a PR to change this to `$too_big`.

Reply:
> we already have to special-case this in some code generation tools for that exact reason. I would support a PR to change this to `$too_big`.

`toobig` for consistence sytel, otherwise, all other enum need change. as all errno does't have underscore.

Reply:
Makes sense to me too. I've now renamed `2big` to `too_big`.

Reply:
I'm not opposed to renaming all the errno values, but each change here does result in a fair amount of manual effort in libc and implementations so my sense is to say is that this could be a separate PR, if someone is interested in seeing this change through. I've now renamed `too_big` to `toobig` to avoid the inconsistency for now.